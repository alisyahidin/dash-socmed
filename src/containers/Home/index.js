import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Container, Row, Col } from 'react-bootstrap'

import withNavbar from '../../hoc/withNavbar'
import Card from '../../components/Card/'
import PostForm from '../../components/PostForm/'
import Post from '../../components/Card/Post/'

import { fetchPost } from '../../actions/post'

class Home extends Component {
  componentDidMount() {
    this.props.fetchPost()
  }

  render() {
    const { post } = this.props

    return (
      <Container>
        <Row>
          <Col md={8}>
            <Card>
              {post.data.map((data, index) => (
                <Post key={index} {...data} />
              ))}
            </Card>
          </Col>
          <Col md={4}>
            <Card>
              <PostForm />
            </Card>
          </Col>
        </Row>
      </Container>
    )
  }
}

const mapStateToProps = state => ({ ...state })

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    fetchPost
  }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(withNavbar(Home))