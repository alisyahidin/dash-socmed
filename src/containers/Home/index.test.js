import React from 'react'
import ReactDOM from 'react-dom'
import Home from './index'

it('render Home container', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Home />, div)
  ReactDOM.unmountComponentAtNode(div)
})
