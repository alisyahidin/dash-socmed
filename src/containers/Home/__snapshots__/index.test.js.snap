// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Home /> correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
